subprojects/build-init/src/integTest/groovy/org/gradle/buildinit/plugins/BuildInitPluginIntegrationTest.groovy

```diff
@@ -276,6 +276,11 @@ include("child")
                     groovy
                     kotlin
 
+     --modularize     Should the build include library projects?
```

Comment:
I think this might have the same issues with confusion over what "modularize" means.  What about `--with-libraries`?  It would just be a boolean flag.  By default, everything is put into one project, but with this flag you get functionality split into libraries instead.

Reply:
I would like to add an additional option here in the future to also generate the projects as Java modules to replace the samples here: https://docs.gradle.org/current/samples/index.html#java_modules

So I would like to keep this open for more choice and not use a boolean. I see how "modularize" might be confusing as the term is so overloaded, but for me it captures what this is about (and I have no better idea). I the options make it clear what this is about.

Reply:
Maybe following the discussion below we could call this something like `--split-project`

Reply:
I think that might be another dimension or question altogether. I think you could have a monolithic application that uses the module system and a multi-project application+libraries that doesn't.  I think module system/no module system is more of a question about the runtime you're going to expect.

I still like `--with-libraries` if it's a simple boolean.  Would something like `--structure=[mono,multi]` work?

Naming things is hard :/

Reply:
Lets make it a boolean then and deal with Java Modules ones/if we get to it. `--split-project` ? (to be in line with the question?)

Follow up: https://github.com/gradle/gradle/pull/14475/files#diff-4185a5b289ff9504f1c3da6c74b1044f